pipeline/transformer/transformer_common/src/main/scala/weco/catalogue/transformer/TransformerWorker.scala

```diff
@@ -119,22 +117,58 @@ trait TransformerWorker[Payload <: SourcePayload, SourceData, SenderDest]
       // nightly Sierra reharvest, when ~250k Sierra source records get synced with
       // Calm.  The records get a new modifiedDate from Sierra, but none of the data
       // we care about for the pipeline is changed.
-      case Right((newWork, key)) =>
+      case Right((transformedWork, key)) =>
         retriever
-          .apply(workIndexable.id(newWork))
+          .apply(workIndexable.id(transformedWork))
           .map { storedWork =>
-            if (newWork.shouldReplace(storedWork)) {
-              Right(Some((newWork, key)))
+            val shouldSend = {
+              if (transformedWork.version < storedWork.version) {
+                debug(
+                  s"${transformedWork.id}: transformed Work is older than the stored Work")
+                false
+              }
+
+              // If the new work and the stored work have the same version and the
+              // same data, it's possible this is SQS retrying a message that failed
+              // last time.  e.g. we indexed the work but didn't send an ID to SNS.
+              //
+              // Send the work, just to be sure it got through the pipeline.
+              else if (storedWork.version == transformedWork.version) {
+                debug(
+                  s"${transformedWork.id}: transformed Work and stored Work have the same version")
+                true
+              } else if (storedWork.data != transformedWork.data) {
+                println(
```

Comment:
println?

Reply:
Arrgh. I had this in to turn stuff on/off during tests, didn't mean to check it in. Should be debug logs like the others.